types/wlr_surface.c

```diff
@@ -293,6 +293,7 @@ void wlr_surface_flush_damage(struct wlr_surface *surface) {
 	pixman_region32_init(&surface->current.buffer_damage);
 
 	wl_resource_queue_event(surface->current.buffer, WL_BUFFER_RELEASE);
+	surface->current.buffer = NULL;
```

Comment:
Probably ok for a temporary workaround.
But keep in mind that this will lead to issues as soon as we implement the TODO at the top of the function, i.e. empty the texture when there was a null buffer attached. It would be triggered from this i guess (when damage_flush is called multiple times).